test/e2e/tests/multiple-transactions.spec.js

```diff
@@ -95,4 +101,125 @@ describe('Confirm transactions', function () {
       },
     );
   });
+
+  it('creates multiple queued transactions, then confirms', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        // initiates a transaction from the dapp
+        await openDapp(driver);
+        // creates first transaction
+        await driver.clickElement({
+          text: 'Send EIP 1559 Transaction',
+          tag: 'button',
+        });
+        await driver.waitUntilXWindowHandles(3);
+        const windowHandles = await driver.getAllWindowHandles();
+        const extension = windowHandles[0];
+        const confirmation = windowHandles[2];
+        await driver.switchToWindowWithTitle('E2E Test Dapp', windowHandles);
+
+        // creates second transaction
+        await driver.clickElement({
+          text: 'Send EIP 1559 Transaction',
+          tag: 'button',
+        });
+        await driver.switchToWindow(confirmation);
+
+        // confirms second transaction
+        await driver.waitForSelector({
+          text: 'Reject 2 transactions',
+          tag: 'a',
+        });
+        await driver.clickElement({ text: 'Confirm', tag: 'button' });
+        await driver.waitForElementNotPresent('.loading-overlay__spinner');
+        // confirms first transaction
+        await driver.clickElement({ text: 'Confirm', tag: 'button' });
+
+        await driver.waitUntilXWindowHandles(2);
+        await driver.switchToWindow(extension);
+        await driver.delay(regularDelayMs);
+        await driver.clickElement('[data-testid="home__activity-tab"]');
+
+        await driver.wait(async () => {
```

Comment:
Do we need a custom `wait` here? Is it worth adding an optional timeout to `findElements` like `waitForSelector` has?

Reply:
I've run some tests in the pipeline and locally and it can be simplified by removing the custom await.